src/sd/mmc.rs

```diff
@@ -0,0 +1,448 @@
+        _d3: Option<impl Peripheral<P = impl InputPin + OutputPin> + 'd>,
+        _d4: Option<impl Peripheral<P = impl InputPin + OutputPin> + 'd>,
+        _d5: Option<impl Peripheral<P = impl InputPin + OutputPin> + 'd>,
+        _d6: Option<impl Peripheral<P = impl InputPin + OutputPin> + 'd>,
+        _d7: Option<impl Peripheral<P = impl InputPin + OutputPin> + 'd>,
+        cd: Option<impl Peripheral<P = impl InputPin> + 'd>,
+        wp: Option<impl Peripheral<P = impl InputPin> + 'd>,
+    ) -> Result<Self, EspError> {
+        let slot_config = sdmmc_slot_config_t {
+            width: width as _,
+            flags: if internal_pullups {
+                SDMMC_INTERNAL_PULLUPS_ENABLE_FLAG
+            } else {
+                0
+            },
+            __bindgen_anon_1: sdmmc_slot_config_t__bindgen_ty_1 {
+                cd: cd
+                    .map(|cd| cd.into_ref().deref().pin())
+                    .unwrap_or(SDMMC_SLOT_NO_CD),
+            },
+            __bindgen_anon_2: sdmmc_slot_config_t__bindgen_ty_2 {
+                wp: wp
+                    .map(|wp| wp.into_ref().deref().pin())
+                    .unwrap_or(SDMMC_SLOT_NO_WP),
+            },
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            clk: _clk.into_ref().deref().pin(),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            cmd: _cmd.into_ref().deref().pin(),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d0: _d0.into_ref().deref().pin(),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d1: _d1.map(|d1| d1.into_ref().deref().pin()).unwrap_or(-1),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d2: _d2.map(|d2| d2.into_ref().deref().pin()).unwrap_or(-1),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d3: _d3.map(|d3| d3.into_ref().deref().pin()).unwrap_or(-1),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d4: _d4.map(|d4| d4.into_ref().deref().pin()).unwrap_or(-1),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d5: _d5.map(|d5| d5.into_ref().deref().pin()).unwrap_or(-1),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d6: _d6.map(|d6| d6.into_ref().deref().pin()).unwrap_or(-1),
+            #[cfg(esp_idf_soc_sdmmc_use_gpio_matrix)]
+            d7: _d7.map(|d7| d7.into_ref().deref().pin()).unwrap_or(-1),
+        };
+
+        {
+            let _cs = USED_SLOTS_CS.enter();
+
+            if USED_SLOTS.load(Ordering::SeqCst) == 0 {
+                esp!(unsafe { sdmmc_host_init() })?;
+
+                USED_SLOTS.fetch_add(1, Ordering::SeqCst);
+            }
+        }
+
+        esp!(unsafe { sdmmc_host_init_slot(S::slot() as _, &slot_config) })?;
+
+        Ok(Self {
+            slot: S::slot(),
+            width,
+            _p: PhantomData,
+        })
+    }
+
+    pub(crate) fn slot_no(&self) -> u8 {
```

Comment:
no = number?

Reply:
I think no is an accepted abbreviation for number? But to be less contradictory, I'll change to num or will just remove the suffix.